airflow/providers/microsoft/azure/hooks/synapse.py

```diff
@@ -0,0 +1,198 @@
+        except KeyError:
+            raise ValueError("A Subscription ID is required to connect to Azure Synapse.")
+
+        credential: Credentials
+        if conn.login is not None and conn.password is not None:
+            if not tenant:
+                raise ValueError("A Tenant ID is required when authenticating with Client ID and Secret.")
+
+            credential = ClientSecretCredential(
+                client_id=conn.login, client_secret=conn.password, tenant_id=tenant
+            )
+        else:
+            credential = DefaultAzureCredential()
+
+        self._conn = self._create_client(credential, conn.host, spark_pool, livy_api_version, subscription_id)
+
+        return self._conn
+
+    @staticmethod
+    def _create_client(credential: Credentials, host, spark_pool, livy_api_version, subscription_id: str):
+        return SparkClient(
+            credential=credential,
+            endpoint=host,
+            spark_pool_name=spark_pool,
+            livy_api_version=livy_api_version,
+            subscription_id=subscription_id,
+        )
+
+    def run_spark_job(
+        self,
+        payload: SparkBatchJobOptions,
+    ):
+        """
+        Run a job in an Apache Spark pool.
+
+        :param payload: Livy compatible payload which represents the spark job that a user wants to submit.
+        """
+        job = self.get_conn().spark_batch.create_spark_batch_job(payload)
+        self.job_id = job.id
+        return job
+
+    def get_job_run_status(
+        self,
+        job_id: int,
+    ):
+        """
+        Get the job run status.
+
+        :param job_id: The job identifier.
+        """
+        job_run_status = self.get_conn().spark_batch.get_spark_batch_job(batch_id=job_id).state
+        return job_run_status
+
+    def wait_for_job_run_status(
+        self,
+        job_id: int,
+        expected_statuses: Union[str, Set[str]],
+        check_interval: int = 60,
+        timeout: int = 60 * 60 * 24 * 7,
+    ) -> bool:
+        """
+        Waits for a job run to match an expected status.
+
+        :param job_id: The job run identifier.
+        :param expected_statuses: The desired status(es) to check against a job run's current status.
+        :param check_interval: Time in seconds to check on a job run's status.
+        :param timeout: Time in seconds to wait for a job to reach a terminal status or the expected
+            status.
+        """
+        job_run_status = self.get_job_run_status(job_id)
+        start_time = time.monotonic()
+
+        while (
+            job_run_status not in AzureSynapseSparkBatchRunStatus.TERMINAL_STATUSES
+            and job_run_status not in expected_statuses
+        ):
+            # Check if the job-run duration has exceeded the ``timeout`` configured.
+            if start_time + timeout < time.monotonic():
+                raise Exception(f"Job {job_id} has not reached a terminal status after {timeout} seconds.")
```

Comment:
```suggestion
                raise AirflowTaskTimeout(f"Job {job_id} has not reached a terminal status after {timeout} seconds.")
```
WDYT about using `AirflowTaskTimeout` exception instead?

Reply:
Nice suggestion! Thank you @josh-fell , I have changed it to use `AirflowTaskTimeout`